src/test/java/rx/AbstractProducerTest.java

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright 2014 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package rx;
+
+import static org.junit.Assert.*;
+
+import java.util.*;
+
+import org.junit.Test;
+
+import rx.Observable.ProducerState;
+import rx.functions.*;
+import rx.observers.TestSubscriber;
+import rx.subscriptions.BooleanSubscription;
+
+public class AbstractProducerTest {
+    @Test
+    public void testJust() {
+        Observable<Integer> source = Observable.create(new Action1<ProducerState<Integer, Void>>() {
+            @Override
+            public void call(ProducerState<Integer, Void> ps) {
+                Subscriber<? super Integer> s = ps.subscriber();
+                s.onNext(1);
+                s.onCompleted();
+                ps.unsubscribe();
```

Comment:
But shouldn't the unsubscribe get invoked immediately anyways due to the Rx contract and `SafeSubscriber` while will trigger an unsubscribe at the end of onCompleted?